R/parallel_new.R

```diff
@@ -248,3 +248,115 @@ choose_framework <- function(
 #' 2008: Proceedings in Computational Statistics. Physica-Verlag HD, 2008.
 #' @name parallelism
 NULL
+
+# ------------------------------------------------------------------------------
+# Choosing how to execute the looping structure
+
+update_parallel_over <- function(control, resamples, grid) {
+  num_candidates <- nrow(grid)
+
+  if (is.null(control$parallel_over) | num_candidates == 0) {
+    control$parallel_over <- "resamples"
+  }
+  if (length(resamples$splits) == 1 & num_candidates > 0) {
+    control$parallel_over <- "everything"
+  }
+  control
+}
+
+# mirai_map() acts a little different form map(), lapply(), etc. It requires
+# that the elements in .args be the args (not symbols). It also requires an
+# extra step to collect the results and coerce them into a list.
+eval_mirai <- function(.x, .f, ..., .args) {
+  .args <- lapply(.args, eval, envir = parent.frame())
```

Comment:
```suggestion
  .args <- lapply(.args, get)
```
`get()` should be marginally faster than `eval()` here.